pyradiosky/skymodel.py

```diff
@@ -1127,6 +1166,277 @@ def point_to_healpix(
+        They can be recalculated afterwards if desired using the `update_positions`
+        method.
+
+        Parameters
+        ----------
+        nside : int
+            nside of healpix map to convert to.
+        order : str
+            Order convention of healpix map to convert to, either "ring" or "nested".
+        to_k : bool
+            Option to convert to K compatible units.
+        full_sky : bool
+            Option to create a full sky healpix map with zeros in the stokes array
+            for pixels with no sources assigned to them. If False only pixels with
+            sources mapped to them will be included in the object.
+        run_check : bool
+            Option to check for the existence and proper shapes of parameters
+            after downselecting data on this object (the default is True,
+            meaning the check will be run).
+        check_extra : bool
+            Option to check optional parameters as well as required ones (the
+            default is True, meaning the optional parameters will be checked).
+        run_check_acceptability : bool
+            Option to check acceptable range of the values of parameters after
+            downselecting data on this object (the default is True, meaning the
+            acceptable range check will be done).
+
+        """
+        if self.component_type != "point":
+            raise ValueError(
+                "This method can only be called if component_type is 'point'."
+            )
+
+        try:
+            import astropy_healpix
+        except ImportError as e:
+            raise ImportError(
+                "The astropy-healpix module must be installed to use HEALPix methods"
+            ) from e
+
+        # clear time & position specific parameters
+        self.clear_time_position_specific_params()
+
+        # `skycoord_to_healpix` should be used in the future when we support any
+        # astropy frame. For now using `lonlat_to_healpix`.
+        hpx_obj = astropy_healpix.HEALPix(nside, order=order)
+        hpx_inds = hpx_obj.lonlat_to_healpix(self.ra, self.dec)
+
+        self._set_component_type_params("healpix")
+        self.nside = nside
+        self.hpx_order = order
+        # now check for duplicates. If they exist, sum the flux in them
+        # if other parameters have variable values, raise appropriate errors
+        if self.hpx_inds.size > np.unique(self.hpx_inds).size:
+            ind_dict = {}
+            for ind in self.hpx_inds:
+                if ind in ind_dict.keys():
+                    pass
+                ind_dict[ind] = np.nonzero(self.hpx_inds == ind)[0]
+                for param in self.ncomponent_length_params:
+                    attr = getattr(self, param)
+                    if attr.value is not None:
+                        param_name = attr.name
+                        if param in ["_reference_frequency", "_spectra_index"]:
```

Comment:
probably a typo in `_spectral_index`

Reply:
Also, it should be possible to combine sources with different reference frequencies ~easily (but not spectral indices).

Reply:
Thanks for the typo catch. I guess I expect that it's much more likely that sources will have the same reference frequency than that they will have the same spectral index, but I can add that logic.

Reply:
hmm, working on this, I'm not sure what to do with the stokes error parameter if the reference frequencies are different. Should the errors scale in the same way that the stokes values do? Or not scale? Or should it error in this case?

Reply:
Oh, I have no idea! What does the stokes error represent?

Reply:
The error bar on the stokes value. GLEAM reports this for instance.

Reply:
I'm naively thinking the errors could just be added in quadrature?

Reply:
That's exactly what I'm doing. What I'm not sure is what to do when they have different reference frequencies.

Reply:
Is the idea of the stokes error that it is the error as defined at the reference frequency, and that it should scale with the spectral index?

Reply:
I don't know, that's what I'm trying to figure out. If I have an object that has stokes_error values and it has components that go to the same healpix pixel that have different reference frequencies, what should I do with the stokes_error values? Should I scale them in the way I do the stokes before I add them in quadrature? Should I just add them in quadrature? Should I error? This is all opened up by trying to accommodate different reference frequencies on component that go into the same pixel. This can of worms is making me feel like going back to just erroring if the reference frequencies are different.

Reply:
Hmm, yeah. Maybe it's not worth the effort -- probably mostly people will not convert to healpix from a catalogue with different ref freqs anyway. We can always revisit later if there's demand.